### ISSUE TITLE
No longer loading sign texts from old schematics

### ISSUE THREAD

**Issue by Brokkonaut** [EngineHub/WorldEdit#2370]
### WorldEdit Version

7.2.16-SNAPSHOT (build 6489)

### Platform Version

git-Paper-"b533905" (MC: 1.20.1)

### Confirmations

- [X] I am using the most recent Minecraft release.
- [X] I am using a version of WorldEdit compatible with my Minecraft version.
- [X] I am using the latest or recommended version of my platform software.
- [X] I am NOT using a hybrid server, e.g. a server that combines Bukkit and Forge. Examples include Arclight, Mohist, and Cardboard.
- [X] I am NOT using a fork of WorldEdit, such as FastAsyncWorldEdit (FAWE) or AsyncWorldEdit (AWE)

### Bug Description

WorldEdit is no longer loading texts in signs in old schematics, this was working in 1.19.4 but no longer in 1.20

### Expected Behavior

The text on those signs should be loaded

### Reproduction Steps

1. Copy the schematic from https://webinterface.cubeside.de/temp/shop_mi_016.schematic to your schematics folder
2. Load, Paste
3. No text on the sign in 1.20

**Comment by emilyy-dev** [comment]
Duplicate of #2369

**Comment by octylFractal** [comment]
I don't think so, I think this is the other way around. We should aim to load older schematics even if the format changes.

**Comment by wizjany** [comment]
old schematics are manually upgraded via the compat handlers at the moment. those only move things to 1.13-compatible formats. moving 1.13->1.20 and beyond would be another step. i tried doing it with DFU when i originally implemented that but it was pretty buggy at the time (well, it still is tbh). paper's dataconverters might alleviate that though, but idk if there's any interest in fixing that atm. this is kinda just how it goes when mojang changes data formats.

**Comment by Brokkonaut** [comment]
I tried this to load schematics: https://github.com/Brokkonaut/WorldEdit/commit/c3adf1b3c1a83791ab0b7fe943d9e08b5050d2f1 - this converts the signs correctly and at least for some old schematics we have I could not find any issues.

I used Constants.DATA_VERSION_MC_1_13_2, because the compatibility handlers already update it to 1.13 and with -1 it just caused errors.

Would it be ok, if I PR it?

**Comment by wizjany** [comment]
I'm a little bit worried that compat handlers are being applied *after* dfu there. if the issues (which led to that being commented out in the first place) were purely in the legacy fixers, then having dfu before compat handlers means things that changed between legacy and 1.13 might have issues still.

**Comment by Brokkonaut** [comment]
For blocks the order was correct (first compatibility handlers, then dfu). For entities it was incorrect and did not work. I did some testing and as far as i can see entity updating works good with the dfu and without compatibility handlers at all: https://github.com/Brokkonaut/WorldEdit/commit/5e92020865bbc3d199dacc6d7c0dcf9a11760885
with this change some long existing bugs when loading old schematics are now fixed. for example paintings are now placed correctly and entities that contain itemstacks now load properly (for example player skulls on armor stands)
